packages/thumbnail-grid-block/src/components/Thumbnail/Item.tsx

```diff
@@ -75,14 +110,14 @@ export const Item = forwardRef<HTMLDivElement, ThumbnailItemProps>(
                             : [],
                         [
                             {
-                                title: 'Replace with upload',
+                                title: image ? 'Replace with upload' : 'Upload image',
                                 icon: <IconArrowCircleUp20 />,
-                                onClick: onOpenFileDialog,
+                                onClick: openFileDialog,
                             },
                             {
-                                title: 'Replace with asset',
+                                title: image ? 'Replace with asset' : 'Choose image',
```

Comment:
```suggestion
                                title: image ? 'Replace with asset' : 'Browse asset',
```

To match the naming on the blockInjection